backend/engine/utils/plugin.py

```diff
@@ -240,7 +240,15 @@ def is_plugin_disabled(settings: dict) -> bool:
     return True
 
 
-def run_plugin(plugin, scan, scan_images, depth=None, include_dev=False, features=None, services=None) -> Result:
+def run_plugin(
```

Comment:
I see some expected data types missing here as well, any reason for that?

Reply:
`scan_images` is left untyped here since I'm not 100% certain all of the places it could be generated -- yet :)

This one is a work-in-progress that hopefully can be properly defined as a dataclass.